### PR TITLE
Fix/societe generale

### PR DESCRIPTION
**Pull request by Romeo-PHILLIPS**
Only thing I wasn't able to implement was using the OpeningHours class, so I made it custom.

Format of opening hours is (closes at mid-day):

```
'openingHoursSpecification': [
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Tuesday',
      'opens': '09:00',
      'closes': '12:35'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Tuesday',
      'opens': '14:00',
      'closes': '17:45'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Wednesday',
      'opens': '09:00',
      'closes': '12:35'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Wednesday',
      'opens': '14:00',
      'closes': '17:45'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Thursday',
      'opens': '09:00',
      'closes': '12:35'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Thursday',
      'opens': '15:00',
      'closes': '17:45'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Friday',
      'opens': '09:00',
      'closes': '12:35'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Friday',
      'opens': '14:00',
      'closes': '17:45'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Saturday',
      'opens': '09:00',
      'closes': '12:45'
    },
    {
      '@type': 'OpeningHoursSpecification',
      'dayOfWeek': 'http://schema.org/Saturday',
      'opens': '14:00',
      'closes': '16:30'
    }
  ],
```
  
  My current output is:
  
  `Tu: 09:00 - 12:35 | 14:00 - 17:45, We: 09:00 - 12:35 | 14:00 - 17:45, Th: 09:00 - 12:35 | 15:00 - 17:45, Fr: 09:00 - 12:35 | 14:00 - 17:45, Sa: 09:00 - 12:45 | 14:00 - 16:30, 
`

### CODE REVIEW

File: locations/spiders/societe_generale.py [hunk 1]
```diff
@@ -8,41 +10,47 @@
...
     name = "societe_generale"
     item_attributes = {"brand": "Societe Generale", "brand_wikidata": "Q270363"}
-    allowed_domains = ["societegenerale.com"]
+    allowed_domains = ["societegenerale.com", "agences.societegenerale.fr"]
     start_urls = [
-        "https://www.societegenerale.com/en/about-us/our-businesses/our-locations",
+        "https://agences.societegenerale.fr/banque-assurance/sitemap_agence_pois.xml",
     ]
 
     def parse(self, response):
```
Comment (Cj-Malone) [inline locations/spiders/societe_generale.py:18]: You can inherit a [SitemapSpider](https://docs.scrapy.org/en/latest/topics/spiders.html#sitemapspider) then you don't have to do this function.

**Comment by Cj-Malone** [comment]
Ideally it should be [OSM format](https://wiki.openstreetmap.org/wiki/Key:opening_hours), you're not far off

`Tu 09:00-12:35,14:00-17:45; We 09:00-12:35,14:00-17:45; Th 09:00-12:35,15:00-17:45; Fr 09:00-12:35,14:00-17:45; Sa 09:00-12:45,14:00-16:30`

File: locations/spiders/societe_generale.py (hunk 1 above)
Comment (Romeo-PHILLIPS) [reply locations/spiders/societe_generale.py:18]: I tried using sitemap_urls (with and without sitemap_rules), but for some reason it doesn't parse anything. Could it be due to the sitemap URL taking a bit to load? https://agences.societegenerale.fr/banque-assurance/sitemap_agence_pois.xml

```
sitemap_urls = (
        "https://agences.societegenerale.fr/banque-assurance/sitemap_agence_pois.xml",
    )
sitemap_rules = [(r"https://agences.societegenerale.fr/banque-assurance/.*", "parse")]

```

`2022-06-30 10:58:11 [scrapy.extensions.logstats] INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)`

Comment (Cj-Malone) [reply locations/spiders/societe_generale.py:16]: See https://github.com/alltheplaces/alltheplaces/pull/3709/commits/267766283fb41a3a2697a7772e96c94043f21bb5.

I don't know exactly what you did, but maybe you didn't rename `parse_store` to `parse`?
By default everything found in the sitemap is passed to `parse`,

**Comment by Cj-Malone** [comment]
LGTM

**Comment by iandees** [comment]
There's still some extra commas in the format: 

```
Tu 09:00,12:20,13:30-18:00; We 09:00,12:20,13:30-18:00; Th 09:00,12:20,14:30-18:00; Fr 09:00,12:20,13:30-18:00; Sa 08:30,12:30
```

This should probably be `Tu 09:00-12:20,13:30-18:00`, etc.

What kept you from using the OpeningHours object? Maybe we should fix that.

**Comment by Cj-Malone** [comment]
I can't talk for @Romeo-PHILLIPS, but I had assumed that class didn't work with multiple hours. 🤦

Anyway, swapped it over now, made a small fix and added a test.

**Comment by iandees** [comment]
Thanks everyone!

**Pull request merged**